pkgs/applications/networking/instant-messengers/ring-daemon/default.nix

```diff
@@ -0,0 +1,133 @@
+{ stdenv
+, lib
+, callPackage
+, fetchgit
+, which
+, pkgconfig
+, autoconf
+, automake
+, libtool
+, pjsip
+, libyamlcpp
+, alsaLib
+, libpulseaudio
+, libsamplerate
+, libsndfile
+, dbus
+, dbus_cplusplus
+, ffmpeg
+, udev
+, pcre
+, gsm
+, speex
+, boost
+, opendht
+, libmsgpack
+, gnutls
+, zlib
+, jsoncpp
+, xlibs
+, libargon2
+, cryptopp
+, openssl
+, perl
+, python3
+, bash
+, libupnp
+, speexdsp
+}:
+
+let
+  python = python3.withPackages (ps: with ps; [
+    pygobject3
+    dbus-python
+  ]);
+  restbed = callPackage ./restbed {};
+  pjsip' = lib.overrideDerivation pjsip (old: {
+    patches = [
+      ./pjsip/gnutls.patch
+      ./pjsip/notestsapps.patch
+      ./pjsip/fix_base64.patch
+      ./pjsip/ipv6.patch
+      ./pjsip/ice_config.patch
+      ./pjsip/multiple_listeners.patch
+      ./pjsip/pj_ice_sess.patch
+      ./pjsip/fix_turn_fallback.patch
+      ./pjsip/fix_ioqueue_ipv6_sendto.patch
+      ./pjsip/add_dtls_transport.patch
+    ];
+    CFLAGS = "-g -DPJ_ICE_MAX_CAND=256 -DPJ_ICE_MAX_CHECKS=150 -DPJ_ICE_COMP_BITS=2 -DPJ_ICE_MAX_STUN=3 -DPJSIP_MAX_PKT_LEN=8000";
+  }); 
+in
+stdenv.mkDerivation rec {
+  name = "dring-${version}";
+  version = "2017-07-11";
+
+  src = fetchgit {
+    url = https://gitlab.savoirfairelinux.com/ring/ring-daemon.git;
+    rev = "006b8dc7be08fe9beb68709af71004e7bc1ceb5c";
+    sha256 = "0ih9g0rismrhx6nqcy3jqfbcs166grg0shnfmrnmykl9h0xy8z47";
+  };
+
+  buildInputs = [
+    which
```

Comment:
A rule of thumb is that tools that are used only as part of the build process (hence only runs on the build machine) can/should be added to the native inputs. Consider the case where build and host differ. In that case, a `which` built for the host would fail to run on the build machine, so if which was intended to be used during the build, you'd need to mark it as native. If which is intended to run on the host (suppose the program you're building shells out to `which` at runtime) you'd add it to `buildInputs`.